### DISCUSSION TITLE
Some images (apart from Google) returning as broken links

### DISCUSSION THREAD

**Discussion by jt196** [searx/searx#2625] [Installation issue: Docker] [answered]
This is what's happening:

![image](https://user-images.githubusercontent.com/8290163/109987808-db6fb280-7cfe-11eb-8708-69195acffea5.png)

It's not showing up on searx.info for example, but I've no idea why it's on my config. If I click on the links, they resolve correctly, they're just not being displayed. 

Been a while since I set this up, just coming back to it as it's a little niggle that is getting a bit annoying. Can't see anything on the basic docker logs, but I don't think they're particularly verbose on searches.

My docker-compose file:

```Docker
version: '3.7'

services:

  filtron:
    container_name: filtron
    image: dalf/filtron
    restart: always
    ports:
      - 4040:4040
      - 4041:4041
    command: -listen 0.0.0.0:4040 -api 0.0.0.0:4041 -target 0.0.0.0:8082
    volumes:
      - ./filtron/:/etc/filtron/
    read_only: true
    cap_drop:
      - ALL
    network_mode: host
    labels:
      - com.centurylinklabs.watchtower.enable=true
    logging:
        driver: loki
        options:
            loki-url: http://localhost:3100/loki/api/v1/push

  searx:
    container_name: searx
    # image: searx/searx:0.17.0-80-a2430154
    image: searx/searx:latest
    restart: always
    command: -f
    volumes:
      - ./searxconf:/etc/searx:rw
    environment:
      - BIND_ADDRESS=0.0.0.0:8080
      - BASE_URL=https://${SEARX_HOSTNAME}/
      - MORTY_URL=https://${SEARX_HOSTNAME}/
      - MORTY_KEY=${MORTY_KEY}
    cap_drop:
      - ALL
    cap_add:
      - CHOWN
      - SETGID
      - SETUID
      - DAC_OVERRIDE
    network_mode: host
    labels:
      - com.centurylinklabs.watchtower.enable=true
    logging:
        driver: loki
        options:
            loki-url: http://localhost:3100/loki/api/v1/push

  morty:
    container_name: morty
    restart: always
    ports:
      - 3001:3001
    environment:
        - DEBUG=false
        - 'MORTY_ADDRESS=0.0.0.0:3001'
        - MORTY_KEY=${MORTY_KEY}
    command: -listen 0.0.0.0:3001 -timeout 6 -ipv6
    volumes:
        - './filtron/rules.json:/etc/filtron/rules.json:rw'
    image: dalf/morty
    network_mode: host
    logging:
      driver: none
    read_only: true
    cap_drop:
      - ALL
    labels:
      - com.centurylinklabs.watchtower.enable=true
    logging:
        driver: loki
        options:
            loki-url: http://localhost:3100/loki/api/v1/push

  searx-checker:
    container_name: searx-checker
    image: searx/searx-checker
    restart: always
    command: -cron -o html/data/status.json http://localhost:8080
    volumes:
      - ./searx-checker:/usr/local/searx-checker/html/data:rw
    network_mode: host
    labels:
      - com.centurylinklabs.watchtower.enable=true
    logging:
        driver: loki
        options:
            loki-url: http://localhost:3100/loki/api/v1/push

volumes:
  searx-checker:
```

**Comment by return42** [comment]
In your browser open developer tools, go to "network" tab and have a look why your browser does not load the images.

Most often it is blocked by the browser due to CSP is enabled by the HTTP server but misses to allow to fetch some data ..

https://github.com/searx/searx/blob/34de715e620a192185e093e1e6caca04ae3cb77a/searx/engines/google_images.py#L9-L12

If this does not help have a look at  #2294

**Comment by jt196** [comment]
Just checked developer tools and it's a 302 error by the looks of it. 

Reading the post you linked, that looks like it's a Morty config IP address issue. Will check out my set up but you're right, I don't think Morty has a public address hooked up to the RP. When I get a moment over the next few days (likely after the weekend) I'll test it out. As soon as it's running and this is confirmed, I'll close the issue. Thanks for the quick response.

**Comment by jt196** [comment, marked as answer]
So, checked this in Firefox and Safari, even a private window and the images were fine. I got it working eventually. I assumed that as it was working in the other browsers it was a cache/preferences thing. 

What **didn't work:**

1. Disabling uBlock Origin
2. Disabling Phishing/Malware protection / DNS to solve navigation errors (Vivaldi)

What **DID work**:

- Going to preferences and disabling image proxy option. 

TBH I thought I did this but obviously not! 

---

The issue you posted mentions CDNs, CSPs, DNS and other acronyms. It also talks about using a reverse proxy and domain with morty as opposed to a local address which is what I've got. Didn't bother with those.

**Comment by jt196** [comment]
I would suggest this is closed, but I can't close it myself as I it's been moved and I'm no longer the author.

**Reply by unixfox** [reply]
There is no concept of closing a discussion.

You can select an answer that solved your question (which I did for you) and that's it.

**Reply by jt196** [reply]
OK thanks - not that familiar with the Github discussions. Cheers.
